### PR TITLE
Openshift support - Update helm chart and bump version

### PR DESCRIPTION
**Pull request by nishantapatil3**
| Q               | A
| --------------- | ---
| Bug fix?        | no
| New feature?    | yes (Openshift deployment support)
| API breaks?     | no
| Deprecations?   | no
| Related tickets | fixes deployment in RHOS-4.11
| License         | Apache 2.0

### What's in this PR?
1. Update `podSecurityContext` and `securityContext` to support deployment in Redhat Openshift 4.11(k8s 1.24+)
2. Replace `nobody:nobody` user to non-root user `appuser:1000` and `appgroup:1000`
3. update `LICENSEI_VERSION = 0.7.0`and `GOLANGCI_VERSION ?= 1.45.2`
4. Update controller_gen and fix download script

### Why?
This PR fixes requirements for restricted mode deployment of pod on RHOS-4.11+ by updating the minimum required openshift policy settings for imps-controller

### Additional context
Openshift requires additional policy settings for imps-controller to operate. This change addresses the requirement to deploy imp-controller on both K8s and RHOS clusters

### Checklist
- [x] Code meets the [Developer Guide](https://github.com/banzaicloud/developer-guide)
- [x] User guide and development docs updated (no doc update necessary)
- [x] Related Helm chart(s) updated (if needed)

### Manual Tests
This PR change is tested to work on both Generic K8s - 1.23+, clusters and RHOS clusters - 4.11+

#### Kubernetes cluster
1. k8s 1.23.X
```
❯ k get node
NAME                                           STATUS   ROLES    AGE    VERSION
ip-192-168-64-65.us-west-2.compute.internal    Ready    <none>   4d6h   v1.23.13-eks-fb459a0
ip-192-168-73-114.us-west-2.compute.internal   Ready    <none>   3d6h   v1.23.13-eks-fb459a0
ip-192-168-74-119.us-west-2.compute.internal   Ready    <none>   4d     v1.23.13-eks-fb459a0
❯
❯ k get deployments.apps -n smm-registry-access
NAME                          READY   UP-TO-DATE   AVAILABLE   AGE
imagepullsecrets-controller   1/1     1            1           26s
```

2. k8s 1.24.x
```
❯ k get node
NAME                                           STATUS   ROLES    AGE   VERSION
ip-192-168-71-118.us-west-2.compute.internal   Ready    <none>   22h   v1.24.7-eks-fb459a0
ip-192-168-71-133.us-west-2.compute.internal   Ready    <none>   22h   v1.24.7-eks-fb459a0
ip-192-168-79-43.us-west-2.compute.internal    Ready    <none>   22h   v1.24.7-eks-fb459a0
❯
❯ k get deployments.apps -n smm-registry-access
NAME                          READY   UP-TO-DATE   AVAILABLE   AGE
imagepullsecrets-controller   1/1     1            1           11s
❯
```

#### Redhat Openshift cluster
1. RHOS - 4.11
```
❯ k get node
NAME                                         STATUS   ROLES          AGE   VERSION
ip-10-0-134-243.us-west-2.compute.internal   Ready    master         11d   v1.24.6+5658434
ip-10-0-174-81.us-west-2.compute.internal    Ready    master         11d   v1.24.6+5658434
ip-10-0-182-132.us-west-2.compute.internal   Ready    infra,worker   11d   v1.24.6+5658434
ip-10-0-202-210.us-west-2.compute.internal   Ready    worker         11d   v1.24.6+5658434
ip-10-0-210-110.us-west-2.compute.internal   Ready    master         11d   v1.24.6+5658434
ip-10-0-210-214.us-west-2.compute.internal   Ready    infra,worker   11d   v1.24.6+5658434
ip-10-0-225-3.us-west-2.compute.internal     Ready    worker         11d   v1.24.6+5658434
❯
❯ k get deployments.apps -n smm-registry-access
NAME                          READY   UP-TO-DATE   AVAILABLE   AGE
imagepullsecrets-controller   1/1     1            1           2m53s
```

2. RHOS - 4.12
```
❯ k get nodes
NAME                                         STATUS   ROLES                  AGE    VERSION
ip-10-0-130-167.us-west-2.compute.internal   Ready    worker                 131m   v1.25.4+77bec7a
ip-10-0-164-101.us-west-2.compute.internal   Ready    infra,worker           110m   v1.25.4+77bec7a
ip-10-0-216-34.us-west-2.compute.internal    Ready    control-plane,master   140m   v1.25.4+77bec7a
ip-10-0-217-167.us-west-2.compute.internal   Ready    control-plane,master   140m   v1.25.4+77bec7a
ip-10-0-218-226.us-west-2.compute.internal   Ready    worker                 130m   v1.25.4+77bec7a
ip-10-0-241-150.us-west-2.compute.internal   Ready    control-plane,master   140m   v1.25.4+77bec7a
ip-10-0-243-77.us-west-2.compute.internal    Ready    infra,worker           110m   v1.25.4+77bec7a
❯
❯ k get deployments.apps -n smm-registry-access
NAME                          READY   UP-TO-DATE   AVAILABLE   AGE
imagepullsecrets-controller   1/1     1            1           92s
```

### CODE REVIEW

**Comment by nishantapatil3** [comment]
Tested on RHOS 4.10
Works after Configuring seccomp profiles as given in this page https://docs.openshift.com/container-platform/4.9/security/seccomp-profiles.html#configuring-default-seccomp-profile_configuring-seccomp-profiles
```
❯ k get nodes
NAME                                         STATUS   ROLES          AGE     VERSION
ip-10-0-132-62.us-west-2.compute.internal    Ready    master         8h      v1.23.12+8a6bfe4
ip-10-0-143-87.us-west-2.compute.internal    Ready    worker         8h      v1.23.12+8a6bfe4
ip-10-0-179-30.us-west-2.compute.internal    Ready    master         8h      v1.23.12+8a6bfe4
ip-10-0-189-163.us-west-2.compute.internal   Ready    infra,worker   7h48m   v1.23.12+8a6bfe4
ip-10-0-214-48.us-west-2.compute.internal    Ready    master         8h      v1.23.12+8a6bfe4
ip-10-0-226-111.us-west-2.compute.internal   Ready    infra,worker   7h48m   v1.23.12+8a6bfe4
ip-10-0-229-144.us-west-2.compute.internal   Ready    worker         8h      v1.23.12+8a6bfe4
❯
❯ k get deployments.apps -n smm-registry-access
NAME                          READY   UP-TO-DATE   AVAILABLE   AGE
imagepullsecrets-controller   1/1     1            1           19m
```

Without Configuring seccomp profiles, replicaset fails with below error
```
│ Events:                                                                                                                                                                                                            │
│   Type     Reason            Age                    From                   Message                                                                                                                                 │
│   ----     ------            ----                   ----                   -------                                                                                                                                 │
│   Warning  FailedCreate      7m24s                  replicaset-controller  Error creating: pods "imagepullsecrets-controller-848956578b-q98s4" is forbidden: unable to validate against any security context const │
│ raint: [pod.metadata.annotations.seccomp.security.alpha.kubernetes.io/pod: Forbidden: seccomp may not be set pod.metadata.annotations.container.seccomp.security.alpha.kubernetes.io/controller: Forbidden: seccom │
│ p may not be set provider "anyuid": Forbidden: not usable by user or serviceaccount provider "nonroot": Forbidden: not usable by user or serviceaccount provider "pcap-dedicated-admins": Forbidden: not usable by │
│  user or serviceaccount provider "hostmount-anyuid": Forbidden: not usable by user or serviceaccount provider "machine-api-termination-handler": Forbidden: not usable by user or serviceaccount provider "hostnet │
│ work": Forbidden: not usable by user or serviceaccount provider "hostaccess": Forbidden: not usable by user or serviceaccount provider "splunkforwarder": Forbidden: not usable by user or serviceaccount provider │
│  "node-exporter": Forbidden: not usable by user or serviceaccount provider "privileged": Forbidden: not usable by user or serviceaccount]
```

**Comment by nishantapatil3** [comment]
@Laci21 please review when you get time, thanks

**Review by Laci21 — APPROVED** [review]
lgtm

**Comment by nishantapatil3** [comment]
Thanks for the review @Laci21

File: Dockerfile [hunk 1]
```diff
@@ -1,9 +1,17 @@
...
+ARG GID=1000
+ARG UID=1000
 
 # Build the manager binary
 FROM golang:${GO_VERSION}-alpine3.14 as builder
+ARG GID
+ARG UID
+
+# Create user and group
+RUN addgroup -g ${GID} -S appgroup && \
```
[the remark this answers is not in the document]
Comment (nishantapatil3) [reply Dockerfile:13]: `golang:${GO_VERSION}-alpine3.14` seems to use base aline image which doesnt use 1000 uid gid, if it did use then docker-build fails so we can catch that early before commit

**Pull request merged**